keanu-python/keanu/context.py

```diff
@@ -66,15 +67,19 @@ def to_java_map(self, python_map: Dict[Any, Any]) -> JavaMap:
         m = self._gateway.jvm.java.util.HashMap()
 
         for (k, v) in python_map.items():
-            m.put(k.unwrap(), v.unwrap())
+            new_k = k.unwrap() if isinstance(k, JavaObjectWrapper) else k
+            new_v = v.unwrap() if isinstance(v, JavaObjectWrapper) else v
+
+            m.put(new_k, new_v)
 
         return m
 
     def to_java_object_list(self, l: Iterable[Any]) -> JavaList:
         lst = self._gateway.jvm.java.util.ArrayList()
 
         for o in l:
-            lst.add(o.unwrap())
+            o = o.unwrap() if isinstance(o, JavaObjectWrapper) else o
```

Comment:
I don't think you need this change - it's not used in your PR

Reply:
I actually do - the Bayesian network construction fails otherwise since the vertices returned by the getAllVertices() function are JavaObject and not JavaObjectWrapper